zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/ServerAuthenticationProvider.java

```diff
@@ -132,8 +132,7 @@ public int getPerm() {
      */
     public abstract boolean matches(ServerObjs serverObjs, MatchValues matchValues);
 
-    @Override
-    public final KeeperException.Code handleAuthentication(ServerCnxn cnxn, byte[] authData) {
+    public KeeperException.Code handleAuthentication(ServerCnxn cnxn, byte[] authData) {
```

Comment:
Is there a reason this class needs to be changed?

Reply:
So handleAuthentication method can not be defined as "final" here because we wanted to "override" it's implementation in "X509ZNodeGroupAclProvider"

Reply:
I still don't understand why this change is required.

The following is my understanding:
`X509ZNodeGroupAclProvider` shouldn't implement `handleAuthentication(ServerCnxn cnxn, byte[] authData)`
It only implements `handleAuthentication(ServerObjs serverObjs, byte[] authData)`.

> we wanted to "override" it's implementation in "X509ZNodeGroupAclProvider"
Who's "we" and where is this implementation overridden in `X509ZNodeGroupAclProvider`?

Reply:
Yes i think i misunderstood between both handleAuthentication methods. While trying out things i though it was required but it's not required and hence i reverted it. Thanks for pointing it out :)